### PR TITLE
handle 'LUKS on LVM' and 'LVM on LUKS'

### PR DESCRIPTION
**Pull request by illiliti**
make messages more consistent, minor improvements

### CODE REVIEW

File: lib/init/rc.boot [hunk 1]
```diff
@@ -121,11 +122,15 @@ log "Loading sysctl settings..."; {
     done
 }
 
-log "Killing device manager to make way for service..."; {
+log "Stopping device manager to make way for service..."; {
     if command -v udevd >/dev/null; then
+        log "Stopping udev..."
+
```
Comment (dylanaraps) [inline lib/init/rc.boot]: This message isn't needed.

File: lib/init/rc.boot [hunk 2]
```diff
@@ -121,11 +122,15 @@ log "Loading sysctl settings..."; {
...
         udevadm control --exit
 
-    elif [ "$mdev_pid" ]; then
+    elif command -v mdev >/dev/null; then
+        log "Stopping mdev..."
```
Comment (dylanaraps) [inline lib/init/rc.boot]: This message isn't needed.

File: lib/init/rc.boot [hunk 3]
```diff
@@ -121,11 +122,15 @@ log "Loading sysctl settings..."; {
...
 
-log "Killing device manager to make way for service..."; {
+log "Stopping device manager to make way for service..."; {
     if command -v udevd >/dev/null; then
+        log "Stopping udev..."
+
         udevadm control --exit
 
-    elif [ "$mdev_pid" ]; then
+    elif command -v mdev >/dev/null; then
```
Comment (dylanaraps) [inline lib/init/rc.boot]: We should stick to using the pid.

File: lib/init/rc.lib [hunk 4]
```diff
@@ -16,8 +16,7 @@ mounted() {
     # This is a pure shell mountpoint implementation. We're dealing
     # with basic (and fixed/known) input so this doesn't need to
     # handle more complex cases.
-    [ -e "$1" ]         || return 1
-    [ -e /proc/mounts ] || return 1
+    [ -e "$1" ] && [ -e /proc/mounts ] || return 1
```
Comment (dylanaraps) [inline lib/init/rc.lib]: This was fine as it was.

File: lib/init/rc.lib [hunk 5]
```diff
@@ -129,6 +128,17 @@ random_seed() {
...
+trigger_lvm() {
+    case "$1" in
+        boot)
+            lvm vgchange --sysinit -aay
+        ;;
+        shutdown)
+            lvm vgchange --sysinit -an
+        ;;
+    esac
+}
```
Comment (dylanaraps) [inline lib/init/rc.lib]:
```sh
case $1 in
    boot)     lvm vgchange --sysinit -aay ;;
    shutdown) lvm vgchange --sysinit -an  ;;
esac
```

**Comment by dylanaraps** [comment]
I have removed encryption support in https://github.com/kisslinux/init/commit/23974886810891abd618e236debabe3a8fe5d955. I would be very happy for you to take over development in a separate project (if you are OK with this of course). My rationale for doing so is in the commit message.

I will refrain from releasing a new version of the init scripts until we have a suitable (or equivalent) replacement (or until a reasonable amount of time passes). 

On another note, the lvm2/cryptsetup packages are already Community maintained so this makes a lot of sense to me. 

I'm happy to discuss this further in IRC/email. :+1:

**Comment by illiliti** [comment]
https://github.com/illiliti/kiss-encryption

**Pull request closed without merging**